lab_4_retrieval_w_clustering/main.py

```diff
@@ -287,6 +413,39 @@ def run_single_train_iteration(self) -> list[ClusterDTO]:
         Returns:
             list[ClusterDTO]: List of clusters.
         """
+        for cluster in self.__clusters:
+            cluster.erase_indices()
+
+        document_vectors = self._db.get_vectors()
+        if not document_vectors:
+            raise ValueError("We can't get vectors")
+
+        centroids = [cluster.get_centroid() for cluster in self.__clusters]
+
+        for index, doc in enumerate(document_vectors):
+            distances_to_centroid = [(calculate_distance(doc[1], centroid), ind) for
+                                     ind, centroid in enumerate(centroids)]
+            if any(distance is None or isinstance(distance, bool) for distance, _ in distances_to_centroid):
+                raise ValueError(
```

Comment:
for user there is no difference which function return None, just explain it correctly

Reply:
fixed